shared_model/backend/protobuf/proto_tx_status_factory.hpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PROTO_TX_STATUS_FACTORY_HPP
+#define IROHA_PROTO_TX_STATUS_FACTORY_HPP
+
+#include "backend/protobuf/transaction_responses/proto_tx_response.hpp"
+#include "interfaces/iroha_internal/tx_status_factory.hpp"
+
+namespace shared_model {
+  namespace proto {
+    class ProtoTxStatusFactory : public interface::TxStatusFactory {
+     private:
+      using FactoryReturnType = interface::TxStatusFactory::FactoryReturnType;
+
+     public:
+      // ------------------------| Stateless statuses |-------------------------
+
+      FactoryReturnType makeTxStatusStatelessFail(TransactionHashType,
```

Comment:
These function looks the same. What's the problem in one template method and many instances?

Reply:
Yes, possible but what is the point of it?
We anyway will implement the interface but implementation will be in the header. In current validator design, we want to avoid templates if it possible.